controllers/volumereplicationgroup_controller.go

```diff
@@ -511,6 +520,47 @@ func (v *VRGInstance) listPVsFromS3Store(s3ProfileName string) ([]corev1.Persist
 	)
 }
 
+// sanityCheckPVClusterData returns an error if there are PVs in the input
+// pvList that have conflicting claimRefs that point to the same PVC name but
+// different PVC UID.
+//
+// Under normal circumstances, each PV in the S3 store will point to a unique
+// PVC and the sanity check will succeed.  In the case of failover related
+// split-brain error scenarios, there can be multiple clusters that concurrently
+// have the same VRG in primary state.  During the split-brain scenario, if the
+// VRG is configured to use the same S3 store for both download and upload of
+// cluster data and, if the application added a new PVC to the application on
+// each cluster after failover, the S3 store could end up with multiple PVs for
+// the same PVC because each of the clusters uploaded its unique PV to the S3
+// store, thus resulting in ambiguous PVs for the same PVC.  If the S3 store
+// ends up in such a situation, Ramen cannot determine with certainty which PV
+// among the conflicting PVs should be restored to the cluster, and thus fails
+// the sanity check.
+func (v *VRGInstance) sanityCheckPVClusterData(pvList []corev1.PersistentVolume) error {
+	pvMap := map[string]corev1.PersistentVolume{}
+	// Scan the PVs and create a map of PVs that have conflicting claimRefs
+	for _, thisPV := range pvList {
+		claimName := thisPV.Spec.ClaimRef.Name
```

Comment:
Given that the ClaimRef is Name/Namespace, I would check for Name/Namespace instead of just Name??? Maybe build a key similar to this:
https://github.com/kubernetes/kubernetes/blob/486ca678a03ef1bceb903376bc8f1757a1293fe0/pkg/controller/volume/persistentvolume/index.go#L188

Reply:
Done.